core/src/mindustry/ui/dialogs/MapsDialog.java

```diff
@@ -141,13 +181,41 @@ void setup(){
             i++;
         }
 
-        if(Vars.maps.all().size == 0){
-            maps.add("@maps.none");
+        if(mapList.size == 0){
```

Comment:
`mapList.size` can be non-zero, but the displayed list can still be empty. Add a boolean flag that is set to true after the `continue` in the loop, and check for that.

Reply:
Done